### DISCUSSION TITLE
Is it possible to update already fitted models adding new covariates?

### DISCUSSION THREAD

**Discussion by iparperspective** [inlabru-org/inlabru#224] [General]
Hi,

I am doing a forward model selection to select covariates (rw2 processes) along with a SPDE. Models are relatively big so I was wondering whether I could keep information from a model with N covariates to ease the model fit of a model with N+1 covariates (N referring t the same covariates). 

cmp1 = "y~ Intercept(1) + X1(X1,model="rw2",scale.model=T) + ST(coordinates, model=spde, group...)
cmp2 = "y~ Intercept(1) + X1(X1,model="rw2",scale.model=T) + X2(X2,model="rw2",scale.model=T) + ST(coordinates, model=spde, group...)

I found this https://groups.google.com/g/r-inla-discussion-group/c/Eo2zIEoqTeM/m/Fo1YlHkmAQAJ, which does not do the exact same thing, but it prbably could if we inherit the modes from model 1 "bru1$mode$theta" 

bru1$mode$theta
log(Range) for ST                          log(Stdev) for ST       Log precision for X1 
                          3.911919                           0.455041                           4.592263

and add a "good guess" for the new covariate (X2). 

Log precision for X2 
2.5

Does this make sense at all?
Cheers

**Comment by finnlindgren** [comment]
Yes, you can use the previous mode as starting point for the "common" parameters. The practical issue is how to either expend the joint theta-vector with the new parameters (if you're lucky they will go at the end of the vector, but I'm not sure that's always the case...), or to extract them and use component-specific initial value settings (`comp(..., hyper=list(theta1=list(initial=...),theta2=list(initial=...)))`); the latter doesn't rely on the ordering of the parameters, as long as you extract them by name and not index.

**Comment by iparperspective** [comment]
Thanks Finn!

I see. So using control.mode would require that one makes sure the ordering is correct. Otherwise one can set the inital values in the formula.

Best,
iosu

**Reply by finnlindgren** [reply]
Correct (although in the inlabru system, I'd say "component definitions" instead of "formula", as the latter is a bit ambiguous).
